pkg/execution/state/redis_state/queue_processor.go

```diff
@@ -523,25 +516,32 @@ func (q *queue) scan(ctx context.Context) error {
 			return nil
 		}
 
-		eg := errgroup.Group{}
-
 		// Reduce number of peeked partitions as we're processing multiple accounts in parallel
 		// Note: This is not optimal as some accounts may have fewer partitions than others and
 		// we're leaving capacity on the table. We'll need to find a better way to determine the
 		// optimal peek size in this case.
-		peekSize := int64(math.Round(float64(PartitionPeekMax / int64(len(peekedAccounts)))))
+		// accountPartitionPeekMax := int64(math.Round(float64(PartitionPeekMax / int64(len(peekedAccounts)))))
+		accountPartitionPeekMax := int64(10)
 
 		// Scan and process account partitions in parallel
+		wg := sync.WaitGroup{}
 		for _, account := range peekedAccounts {
 			account := account
-			eg.Go(func() error {
+
+			wg.Add(1)
+			go func(account uuid.UUID) {
+				defer wg.Done()
 				partitionKey := q.u.kg.AccountPartitionIndex(account)
 
-				return q.scanPartition(ctx, partitionKey, peekSize, peekUntil, nil, metricShardName, &account)
-			})
+				if err := q.scanPartition(ctx, partitionKey, accountPartitionPeekMax, peekUntil, nil, metricShardName, &account); err != nil {
```

Comment:
Accounts are processed in parallel to avoid stalling other accounts. For each account, we peek account partitions, which are again processed in parallel.

Reply:
We peek 30 accounts at once, and for each account peek 10 partitions.

When tuning, we prefer peeking more accounts than partitions per account to ensure fairness. For folks with high-throughput requirements, we set aside workers with guaranteed capacity.